core/trino-main/src/test/java/io/trino/sql/planner/optimizations/TestAddExchangesPlans.java

```diff
@@ -784,6 +785,37 @@ public void testJoinNotExactlyPartitioned()
                                                 tableScan("orders"))))));
     }
 
+    // Negative test for use-exact-partitioning when colocated join is enabled (be default)
```

Comment:
When a test has a pre-requisite that some property should be enabled, then we should include an assertion like `verifyDynamicFilteringEnabled` in it which verifies that the assumption is true.